### PR TITLE
Add validty check function to UUri

### PR DESCRIPTION
**Pull request by sophokles73**
Added a function for checking if an existing UUri is a valid
uProtocol URI.

Addresses #209

### CODE REVIEW

**Comment by sophokles73** [comment]
@PLeVasseur is this what you need for uStreamer?

File: src/uri.rs [hunk 1]
```diff
@@ -419,6 +382,77 @@ impl UUri {
...
+                Ok(auth_name)
+            } else {
+                Err(UUriError::validation_error(
+                    "URI's authority name must not exceed 128 characters",
+                ))
+            }
+        }
+    }
+
+    fn verify_major_version(major_version: u32) -> Result<u8, UUriError> {
```
Comment (PLeVasseur) [inline src/uri.rs:416]: :+1: Yup, this is definitely one of the functions that'd be useful to have.

File: src/uri.rs [hunk 2]
```diff
@@ -419,6 +382,77 @@ impl UUri {
...
+    ///   ue_id: 0x1000,
+    ///   ue_version_major: 0x01,
+    ///   resource_id: 0x8100,
+    ///   ..Default::default()
+    /// };
+    /// assert!(uuri.check_validity().is_ok());
+    /// ```
+    pub fn check_validity(&self) -> Result<(), UUriError> {
+        Self::verify_authority(self.authority_name.as_str())?;
+        Self::verify_major_version(self.ue_version_major)?;
```
Comment (PLeVasseur) [inline src/uri.rs:452]: Could you also add a `verify_resource_id()` that'd attempt the conversion to a `u16` and include calling that function in here too?
image

**Review by PLeVasseur — COMMENTED** [review]
Thanks  @sophokles73!

I think this is one half of what I was asking for, please see comments.

File: src/uri.rs (hunk 2 above)
Comment (sophokles73) [reply src/uri.rs:452]: Oh, absolutely. Don't know how that slipped through. Good that we do the reviews ... ;-)

**Comment by sophokles73** [comment]
@PLeVasseur I have added a corresponding check ...

**Review by PLeVasseur — APPROVED** [review]
Thanks Kai! LGTM

**Pull request merged**